src/utils/updateWiki.ts

```diff
@@ -0,0 +1,272 @@
+    while ((fieldMatch = optionFieldsPattern.exec(options![0])) !== null) {
+      const name = fieldMatch[1];
+      const description = fieldMatch[2];
+      extractedOptions.push({ name, description });
+    }
+    await appendFile(wikiPath, `- **Options:** \n`);
+    for (const desc of extractedOptions) {
+      await appendFile(wikiPath, `    - \`\`${desc.name}\`\`: ${desc.description}\n`);
+    }
+  }
+};
+
+const formatSubcommandDetails = async (subcommandDetails: string | undefined) => {
+  let formattedSubcommandDetails = '';
+  if (subcommandDetails === undefined || subcommandDetails === '') {
+    formattedSubcommandDetails = 'None';
+  } else {
+    const regex = /\b(\w+)\s*:/g;
+    const matches = subcommandDetails.match(regex);
+    if (matches) {
+      const subcommandList = matches.map((match) => match.trim().replace(/:$/, '')).join(', ');
+      formattedSubcommandDetails = subcommandList
+        .replace(/'/g, '')
+        .split(', ')
+        .map((subcmd) => `\`${subcmd}\``)
+        .join(', ');
+    }
+  }
+  await appendFile(wikiPath, `- **Subcommands:** ${formattedSubcommandDetails}\n\n`);
+};
+
+// ----------------------------------- END OF UTILITY FUNCTIONS ---------------------------- //
+
+export const updateWiki = async (): Promise<void> => {
+  logger.info({
+    message: 'Updating wiki...',
+  });
+
+  // Refresh COMMAND-WIKI.md
+  await writeFile(wikiPath, '');
+
+  const filesAndDirs = await readdir(commandDetailsDir);
+  const directories = filesAndDirs.filter((file) =>
+    statSync(`${commandDetailsDir}/${file}`).isDirectory(),
+  );
+
+  for (const dir of directories) {
+    if (!compoundCommands.includes(dir)) {
+      await appendFile(wikiPath, `# ${dir.toUpperCase()}\n`);
+      const subDir = `${commandDetailsDir}/${dir}`;
+      const files = await readdir(subDir);
+      for (const file of files) {
+        const filePath = `${subDir}/${file}`;
+        const content = await readFile(filePath, 'utf-8');
+        const match = content.match(commandDetailsPattern);
+        if (match) {
+          const codeSection = match[1];
+
+          // Extract info pieces
+          const name = codeSection.match(namePattern)?.[1];
+          const aliases = codeSection.match(aliasesPattern)?.[1];
+          const description = codeSection.match(descriptionPattern)?.[1];
+          const detailedDescription = codeSection.match(detailedDescriptionPattern)?.[1];
+          const options = codeSection.match(optionsPattern)?.[1].split(', ');
+          const subcommandDetails = codeSection.match(subcommandDetailsPattern)?.[1];
+
+          // Format the info
+          await formatCommandName(name);
+          await formatAliases(aliases);
+          await formatDescription(description);
+          await formatDetailedDescription(detailedDescription);
+          await formatOptions(options);
+          await formatSubcommandDetails(subcommandDetails);
```

Comment:
I think we can move this into an `extractAndFormat` function since it gets repeated a few times